conjure-go-client/httpclient/response_error_decoder_middleware.go

```diff
@@ -81,7 +81,10 @@ func (d restErrorDecoder) DecodeError(resp *http.Response) error {
 	unsafeParams := map[string]interface{}{}
 	if resp.StatusCode >= http.StatusTemporaryRedirect &&
 		resp.StatusCode < http.StatusBadRequest {
-		unsafeParams["location"] = resp.Header.Get("Location")
+		location, err := resp.Location()
+		if err == nil {
+			unsafeParams["location"] = location.String()
+		}
```

Comment:
Uses the `http.Response.Location()` method here allows us to forgo passing in the `useBaseURIOnly` in the `doOnce()` method since `Location()` builds the proper full URI based on the `http.Response.Request` and returned `Location` header.